backend/Warehouse/src/main/java/dk/sdu/controllab/warehouse/service/WarehouseServiceImpl.java

```
@@ -49,24 +50,64 @@ public GetInventoryResponse getInventory() {
     }
 
     @Override
-    public WarehouseInventoryEntry pickItem(int trayId, int amount) {
-        WarehouseInventoryEntry pickRequest = warehouseRepository.findById(trayId).orElseThrow(() -> new WarehouseException("Tray is empty"));
+    public void moveToOutletTray(int trayId) {
+        if (warehouseRepository.findById(-1).isPresent()) throw new WarehouseException("Outlet tray is full");
+        WarehouseInventoryEntry outletRequest = warehouseRepository.findById(trayId).orElseThrow(() -> new WarehouseException("Tray is empty"));
+        WarehouseInventoryEntry outletEntry = new WarehouseInventoryEntry();
+        outletEntry.setItem(outletRequest.getItem());
+        outletEntry.setId(-1);
+        outletEntry.setAmount(outletRequest.getAmount());
+        warehouseRepository.save(outletEntry);
+        warehouseRepository.deleteById(outletRequest.getId());
+        synchronizeInventory();
+    }
+
+    @Override
+    public void returnFromOutlet(){
```

Should throw an error if no entry is found at -1